### ISSUE TITLE
In Haskell mode, pressing `O` leads to `void-function evil-digit-argument-or-evil-beginning-of-line`

### ISSUE THREAD

**Issue by dschrempf** [hlissner/doom-emacs#5769]
### What did you expect to happen?

A line is opened above the cursor.

### What actually happened?

```
Debugger entered--Lisp error: (void-function evil-digit-argument-or-evil-beginning-of-line)
  (evil-digit-argument-or-evil-beginning-of-line)
  +haskell/evil-open-above()
  funcall-interactively(+haskell/evil-open-above)
  call-interactively(+haskell/evil-open-above nil nil)
  command-execute(+haskell/evil-open-above)
```

### Describe your attempts to resolve the issue

I checked the code of `+haskell/evil-open-above`. I could not find the function `evil-digit-argument-or-evil-beginning-of-line`. I don't exactly know what it does so I was unable to replace it.

### Steps to reproduce

1. Open a Haskell file.
2. Press `O` (capitalized o).

### System Information

https://pastebin.com/kbB5gdfC

**Comment by cnr** [comment]
Looks like this is caused by the [recent removal of `evil-digit-argument-or-evil-beginning-of-line` in evil](https://github.com/emacs-evil/evil/pull/1519/files). I'm confused because evil is supposed to be pinned to a much older commit(?) https://github.com/hlissner/doom-emacs/blob/b4cc1c84d8e77e3a5cb2175be4908b6ce46d93a6/modules/editor/evil/packages.el#L4

From local testing, and following what they do in that PR, replacing with `evil-beginning-of-line` seems to work

```diff
-  (evil-digit-argument-or-evil-beginning-of-line)
+  (evil-beginning-of-line)
```

I don't know enough to be sure that's the right fix

---

Relatedly, `:editor multiple-cursors` uses `evil-digit-argument-or-evil-beginning-of-visual-line`, which was also removed

**Comment by hlissner** [comment]
Duplicate of https://github.com/hlissner/doom-emacs/issues/5731 and fixed in 616956ac948def18d3ae12dad73c5b3623cdf5bd.
